packages/synapse-react-client/src/components/TableColumnSchemaEditor/Validators/ColumnModelValidator.ts

```diff
@@ -0,0 +1,145 @@
+
+const columnModelBaseZodSchema = z.object({
+  name: z.string().min(1, { message: 'Name is required' }),
+  columnType: z.nativeEnum(ColumnTypeEnum),
+})
+
+export const jsonSubColumnModelZodSchema = columnModelBaseZodSchema.merge(
+  z.object({
+    jsonPath: z.string().startsWith('$'),
+    facetType: facetTypeSchema,
+  }),
+)
+
+/**
+ * Zod schema to validate a column model. The provided data is coerced and transformed to match the ColumnModel type.
+ * For this reason, form data may be directly passed into this schema parser.
+ *
+ * The parse method will return a ColumnModel where all data is properly formed.
+ */
+export const columnModelZodSchema = columnModelBaseZodSchema
+  .merge(
+    z.object({
+      id: z.string().optional(),
+      defaultValue: z.union([z.string(), z.array(z.any())]).optional(),
+      maximumSize: z
+        .union([optionalStringSchema, z.number()])
+        .pipe(
+          z.coerce
+            .number()
+            .finite()
+            .int()
+            .min(1)
+            .max(MAX_STRING_SIZE)
+            .optional(),
+        ),
+      maximumListLength: z
+        .union([optionalStringSchema, z.number()])
+        .pipe(
+          z.coerce
+            .number()
+            .finite()
+            .int()
+            .min(1)
+            .max(MAX_LIST_LENGTH)
+            .optional(),
+        ),
+      enumValues: z.array(z.coerce.string()).optional(),
+      jsonSubColumns: z.array(jsonSubColumnModelZodSchema).optional(),
+      facetType: facetTypeSchema.optional(),
+    }),
+  )
+  .refine(data => data.maximumSize == null || canHaveSize(data.columnType), {
+    message: 'Size is not allowed for this column type',
+    path: ['maximumSize'],
+  })
+  .refine(
+    data =>
+      data.maximumListLength == null || canHaveMaxListLength(data.columnType),
+    {
+      message: 'Maximum list length is not allowed for this column type',
+      path: ['maximumListLength'],
+    },
+  )
+  .refine(
+    data => {
+      if (data.jsonSubColumns != null) {
+        return data.columnType === ColumnTypeEnum.JSON
+      }
+      return true
+    },
+    {
+      message: 'Only JSON columns can have JSON sub-columns.',
+      path: ['columnType'],
+    },
+  )
+  .transform((data, ctx) => {
+    if (data.defaultValue != null) {
+      // Validate and transform the defaultValue based on the columnType
+      const defaultValueSchema = getDefaultValueValidator(data.columnType)
+      const result = defaultValueSchema.safeParse(data.defaultValue)
+      let transformedValue: string | undefined
+      if (result.success) {
+        transformedValue = result.data
+      } else {
+        result.error.issues.forEach(issue => {
+          ctx.addIssue({ ...issue, path: ['defaultValue'] })
+        })
+        transformedValue = undefined
+      }
+      return { ...data, defaultValue: transformedValue }
+    }
+    return omit(data, 'defaultValue')
+  })
```

Comment:
Now it gets pretty complicated...the schema we use to validate `defaultValue` will depend on the column type. So...

1. Conditionally fetch a schema
2. `safeParse` the data with the schema
3. If the result has errors, pass those errors along
4. If the result doesn't have errors, transform the value and keep going!